src/sentry/models/authprovider.py

```diff
@@ -68,6 +70,20 @@ def handle_async_replication(self, region_name: str, shard_identifier: int) -> N
             auth_provider=serialized, region_name=region_name
         )
 
+    @classmethod
+    def handle_async_deletion(
+        cls,
+        identifier: int,
+        region_name: str,
+        shard_identifier: int,
+        payload: Mapping[str, Any] | None,
+    ) -> None:
+        from sentry.services.hybrid_cloud.replica.service import region_replica_service
+
+        region_replica_service.delete_replicated_auth_provider(
+            auth_provider_id=identifier, region_name=region_name
+        )
```

Comment:
Were we not properly cleaning these replicas up before?

Reply:
We were, but through an outbox that was sent from region -> control. In the region scoped handler, we would remove the replica and do an RPC call to remove the auth_provider from control. With the endpoint being moved to control I thought that using the `ReplicatedModel` hooks was a better solution.